src/EnergyPlus/RoomAirModelManager.cc

```diff
@@ -1635,11 +1647,20 @@ namespace RoomAir {
                     roomAFNNodeHVAC.SupplyFraction = ipsc->rNumericArgs(iEquipArg);
                     roomAFNNodeHVAC.ReturnFraction = ipsc->rNumericArgs(iEquipArg);
 
+                    // get equipment index
+                    int EquipIndex = 0;
+                    for (int thisZoneEquipNum = 1; thisZoneEquipNum <= state.dataZoneEquip->ZoneEquipList(iZone).NumOfEquipTypes;
+                         ++thisZoneEquipNum) {
+                        if (Util::SameString(state.dataZoneEquip->ZoneEquipList(iZone).EquipName(thisZoneEquipNum), roomAFNNodeHVAC.Name)) {
+                            EquipIndex = state.dataZoneEquip->ZoneEquipList(iZone).EquipIndex(thisZoneEquipNum);
```

Comment:
@rraustad Would it work here to save this value, then pass it in to `CheckEquipName` to be used for the equipment types supported by unitarysystem?
```
HVACSystemData * compPointer;
compPointer = state.dataZoneEquip->ZoneEquipList(iZone).compPointer[thisZoneEquipNum];
```
See `SimZoneEquipment` https://github.com/NREL/EnergyPlus/blob/8c14f4b6e126a062954ca64a52d3defdcadca53c/src/EnergyPlus/ZoneEquipmentManager.cc#L3288-L3309

Reply:
It should. I can't say if this pointer is available on the first call to RoomAirModelManager (calling order) but once available the pointer is saved in the index to the equipment in the equipment list (not the component index). If the UnitarySystem (or other types managed by UnitarySystem) is 4th in the equipment list then it is `.compPointer[4]`.

    thisZoneEquipList.compPointer.resize(thisZoneEquipList.NumOfEquipTypes + 1);

    HVACSystemData * compPointer = state.dataZoneEquip->ZoneEquipList(iZone).compPointer[1 <= n <= NumOfEquipTypes];